### PR TITLE
Made a nice scrollable history

### PR DESCRIPTION
**Pull request by rkuenzi-epfl**

### CODE REVIEW

**Comment by rkuenzi-epfl** [comment]
The PR also includes some updates to access the User lists of pictures so codeclimate complains because there is a lot of code duplication because I couldn't get rid of the previous way to access the database.

File: app/src/androidTest/java/com/github/wnder/FirbaseUserDatabaseTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertFalse;
+
+@RunWith(JUnit4.class)
+public class FirbaseUserDatabaseTest {
```
Comment (BoostedBoat) [inline app/src/androidTest/java/com/github/wnder/FirbaseUserDatabaseTest.java:31]: Missing a little "e" here

**Review by JeremyDezalos — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/wnder/user/FirebaseUserDatabase.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+    private final CollectionReference picturesCollection;
+    private final CollectionReference usersCollection;
+    private final Context context;
+
+    @Inject
+    public FirebaseUserDatabase(Context context){
+        picturesCollection = FirebaseFirestore.getInstance().collection("pictures");
+        usersCollection = FirebaseFirestore.getInstance().collection("users");
+        this.context = context;
```
Comment (BoostedBoat) [inline app/src/main/java/com/github/wnder/user/FirebaseUserDatabase.java:33]: Maybe there are methode for this in the Storage class, not sure though

File: app/src/main/java/com/github/wnder/user/FirebaseUserDatabase.java [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    }
+
+    @Override
+    public CompletableFuture<List<String>> getPictureList(User user, String picturesListName) {
+        CompletableFuture<List<String>> cf = new CompletableFuture<>();
+        if(user instanceof GuestUser){
+            cf.complete(new ArrayList<>());
+        } else{
+
+            usersCollection.document(user.getName()).get().addOnSuccessListener(documentSnapshot -> {
```
Comment (BoostedBoat) [inline app/src/main/java/com/github/wnder/user/FirebaseUserDatabase.java:43]: Same as above

**Review by BoostedBoat — APPROVED** [review]
Actually a very nice history

File: app/src/main/java/com/github/wnder/user/FirebaseUserDatabase.java (hunk 2 above)
Comment (rkuenzi-epfl) [reply app/src/main/java/com/github/wnder/user/FirebaseUserDatabase.java:33]: We want to get rid of the storage class so this is normal

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -46,6 +46,13 @@
 
     <!--History Strings-->
     <string name="content_descr_history_image">Previously guessed picture</string>
+    <string name="content_descr_history_location">Location icon</string>
+    <string name="content_descr_history_scoreboard">Scoreboard icon</string>
+    <string name="your_score">Your score :</string>
```
[the remark this answers is not in the document]
Comment (rkuenzi-epfl) [reply app/src/main/res/values/strings.xml:51]: Done

File: app/src/main/res/layout/history_entry.xml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            android:adjustViewBounds="true"
+            android:scaleType="fitCenter"
+            tools:srcCompat="@tools:sample/backgrounds/scenic[0]" />
+
+        <FrameLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_margin="8dp">
+
+            <TableRow
```
[the remark this answers is not in the document]
Comment (rkuenzi-epfl) [reply app/src/main/res/layout/history_entry.xml]: Done

File: app/src/androidTest/java/com/github/wnder/FirbaseUserDatabaseTest.java (hunk 1 above)
Comment (rkuenzi-epfl) [reply app/src/androidTest/java/com/github/wnder/FirbaseUserDatabaseTest.java]: Y

File: app/src/main/java/com/github/wnder/HistoryAdapter.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+import com.github.wnder.scoreboard.ScoreboardActivity;
+import com.github.wnder.user.GlobalUser;
+
+import java.util.ArrayList;
+import java.util.Locale;
+
+public class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.ViewHolder> {
+
+    private ArrayList<String> pictureList;
+    private PicturesDatabase picturesDb;
```
[the remark this answers is not in the document]
Comment (rkuenzi-epfl) [reply app/src/main/java/com/github/wnder/HistoryAdapter.java]: Done

**Review by JeremyDezalos — APPROVED** [review]

**Pull request merged**